devops/scripts/benchmarks/utils/flamegraph.py

```diff
@@ -0,0 +1,436 @@
+            if os.path.exists(folded_file):
+                os.remove(folded_file)
+
+            if not os.path.exists(svg_file) or os.path.getsize(svg_file) == 0:
+                raise RuntimeError(f"Failed to generate flamegraph SVG: {svg_file}")
+
+            log.debug(f"Generated flamegraph: {svg_file}")
+
+            # Create symlink immediately after SVG generation
+            self._create_immediate_symlink(svg_file)
+
+            # Prune old flamegraph directories
+            self._prune_flamegraph_dirs(os.path.dirname(perf_data_file))
+
+            return svg_file
+
+        except Exception as e:
+            # Clean up on failure
+            for temp_file in [folded_file, svg_file]:
+                if os.path.exists(temp_file):
+                    os.remove(temp_file)
+            raise RuntimeError(f"Failed to generate flamegraph for {bench_name}: {e}")
+
+    def _create_immediate_symlink(self, svg_file: str):
+        """
+        Create a symbolic link for the SVG file immediately after generation.
+        This ensures the web interface can access the file right away.
+        """
+        try:
+            # Check if workdir is available
+            if not options.workdir:
+                log.error("No workdir available for immediate symlink creation")
+                return
+
+            # Use the default HTML path relative to the script location
+            script_dir = os.path.dirname(os.path.dirname(__file__))  # Go up from utils/
+            html_path = os.path.join(script_dir, "html")
+
+            if not os.path.exists(html_path):
+                log.error(f"HTML directory not found: {html_path}")
+                return
+
+            # Calculate the relative path of the SVG file from the flamegraphs directory
+            if not svg_file.startswith(self.flamegraphs_dir):
+                log.error(f"SVG file not in expected flamegraphs directory: {svg_file}")
+                return
+
+            rel_path = os.path.relpath(svg_file, self.flamegraphs_dir)
+            target_dir = os.path.join(html_path, "results", "flamegraphs")
+            target_file = os.path.join(target_dir, rel_path)
+
+            # Create target directory structure
+            os.makedirs(os.path.dirname(target_file), exist_ok=True)
+
+            # Remove existing symlink if it exists
+            if os.path.islink(target_file):
+                os.unlink(target_file)
+            elif os.path.exists(target_file):
+                os.remove(target_file)
+
+            # Create the symlink
+            os.symlink(svg_file, target_file)
+            log.debug(f"Created immediate symlink: {target_file} -> {svg_file}")
+
+            # Update the flamegraph manifest for the web interface
+            self._update_flamegraph_manifest(html_path, rel_path, options.save_name)
+
+        except Exception as e:
+            log.debug(f"Failed to create immediate symlink for {svg_file}: {e}")
+
+    def _update_flamegraph_manifest(
```

Comment:
I genuinely don't understand the idea here. data.js is purely an output file. we should not parse it.